### PR TITLE
Supporting user sing up and login.

### LINKED ISSUE #21: Support user register/login
We need to support registration and login in order to prepare the architecture for future users.

### PR DESCRIPTION
**Pull request by AndresMorelos**
Fix #21 

Client: 

### Login
image

### Sing up
image

### Home

image

### CODE REVIEW

**Comment by AndresMorelos** [comment]
@FlavioAandres read carefully and test all please, it is a big change, and have some breaking changes.

File: client/amplify/backend/auth/clientbec096a1/clientbec096a1-cloudformation-template.yml [hunk 1]
```diff
@@ -0,0 +1,433 @@
+AWSTemplateFormatVersion: 2010-09-09
+
```
Comment (FlavioAandres) [inline client/amplify/backend/auth/clientbec096a1/clientbec096a1-cloudformation-template.yml:2]: Hey i think you should not upload those files and add them to the git ignore 

as already they are tracked you should remove those files from git with `git rm `

File: client/src/components/ProfileContainer/ProfileContainer.jsx [hunk 2]
```diff
@@ -2,44 +2,43 @@ import React from "react";
...
+    })
   }
 
-  onSaveCategory = (category)=>{
+  onSaveCategory = (category) => {
     this.setState({
       showSpinningCategoryModal: true
     })
-    axios.post(constants.basepath + constants.routes.categories, {
+    API.post('finances', '/user/categories', {body: {
```
Comment (FlavioAandres) [inline client/src/components/ProfileContainer/ProfileContainer.jsx:24]: i will create a issue to move back the routes to the constants 🤘🏼:D

File: serverless.yml [hunk 3]
```diff
@@ -46,103 +46,155 @@ functions:
...
           cors:
             - enabled: true
             - origin: "*"
             - headers: "Access-Control-Allow-Origin"
+          response:
+            headers:
+              Access-Control-Allow-Origin: "'*'"
+          authorizer:
+            name: authorizer
+            arn: ${file(./config/${opt:stage}.json):USER_POOL_ARN}
```
Comment (FlavioAandres) [inline serverless.yml:63]: I think we can deploy the ARN using the Cloud formation sintax, another issue 🤘🏼

**Review by FlavioAandres — APPROVED** [review]
LGTM!! it does not looks like a breaking change 👏🏼

**Pull request merged**